File: 10_laba/src/by/bntu/fitr/java/inGoslingsFootsteps/Lab10/controller/CustomerAction.java
package by.bntu.fitr.java.inGoslingsFootsteps.Lab10.controller;

import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.entity.AirTicketOfficeContext;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.entity.Customer;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.exception.BookingException;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.logic.FlightController;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.util.UserInput;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.util.exception.InputException;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.view.Printer;
import org.apache.log4j.Logger;

import java.io.IOException;

public class CustomerAction {
    private static final Logger LOG;
    static{
        LOG = Logger.getLogger(by.bntu.fitr.java.inGoslingsFootsteps.Lab10.controller.CustomerAction.class);
    }

    public static  void choicebyCustomer(FlightController flightController, Customer customer, int action) throws IOException {
        try {
            if (action == 1) {
                AirTicketOfficeContext availableFlight = flightController.getAvailableFlight(customer);
                Printer.print(availableFlight.toString());
                int index = UserInput.inputInt("Choose flight");
                flightController.bookTicket(availableFlight.getFlights()[index - 1], customer);
                LOG.info("Ticket is booked");
            }
            else if (action == 2) {
                AirTicketOfficeContext occupiedCars = flightController.getFlightsBookedByCustomer(customer);
                Printer.print(occupiedCars.toString());
                int index = UserInput.inputInt("Choose flight");
                flightController.returnTicket(occupiedCars.getFlights()[index - 1], customer);
                LOG.info("Ticket is returned");
            }
            else{
                throw new InputException("You should enter 1 or 2");}
        }catch (BookingException | InputException ex) {
            Printer.print(ex.getMessage());
            LOG.error(ex.getMessage());
        }
    }
}

File: 10_laba/C##/AirTicketOffice/AirTicketOffice/by/bntu/fitr/java/ingoslingsfootsteps/controller/CustomerAction.cs
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.entity;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.exception;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.logic;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.util;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.util.exception;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.view;
using NLog;
using System;
using System.Collections.Generic;
using System.Text;

namespace AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.controller
{
    public class CustomerAction
    {
        private static readonly Logger logger = LogManager.GetCurrentClassLogger();

        public static void ChoicebyCustomer(FlightController flightController, Customer customer, int action)
        {
            try
            {
                if (action == 1)
                {
                    AirTicketOfficeContext availableFlight = flightController.GetAvailableFlight(customer);
                    Printer.print(availableFlight.ToString());
                    Printer.print("Choose flight");
                    int index = Convert.ToInt32(UserInput.input());
                    flightController.BookTicket(availableFlight.Flights[index - 1], customer);
                    logger.Info("Ticket is booked");
                }
                else if (action == 2)
                {
                    AirTicketOfficeContext occupiedCars = flightController.GetFlightsBookedByCustomer(customer);
                    Printer.print(occupiedCars.ToString());
                    Printer.print("Choose flight");
                    int index = Convert.ToInt32(UserInput.input());
                    flightController.ReturnTicket(occupiedCars.Flights[index - 1], customer);
                    logger.Info("Ticket is booked");
                }
                else
                {
                    throw new InputException("You should enter 1 or 2");
                }
            }
            catch (Exception ex)
            {
                if (ex is BookingException || ex is InputException)
                {
                    System.Console.WriteLine($"Error: {ex.Message}");
                }
                logger.Error(ex.Message);
            }
        }
    }

}
